src/JestProcessManagement/JestProcess.ts

```diff
@@ -35,8 +35,10 @@ export class JestProcess {
         if (--this.keepAliveCounter > 0) {
           this.runner.removeAllListeners()
           this.startRunner()
-        } else if (this.onExitCallback) {
-          this.onExitCallback(this)
+        } else {
```

Comment:
@connectdotz could you check if this condition is now done right?
Looks like without this change `resolve` was never called until `onExitCallback` was also set.

Reply:
good catch 👍